lib/adapters/QUnitAdapter.js

```diff
@@ -82,41 +48,38 @@ export default class QUnitAdapter extends EventEmitter {
       testDetails.runtime = undefined
     }
 
-    testDetails.errors = this.errors
-
     this.testsDetails[qunitTest.testId] = testDetails
   }
 
-  isGlobalModule (module) {
-    return module.name === ''
-  }
-
   createGlobalSuite () {
     var topLevelSuites = []
-    var suites = []
+    var suites
     var globalSuite
+    var modules
     var indexStart, indexEnd
 
-    this.emitGlobal = true
-
     // Access QUnit internals to get all suites and tests, working around
     // missing event data.
 
     // Create the global suite first.
     if (this.QUnit.config.modules.length > 0 &&
-        this.isGlobalModule(this.QUnit.config.modules[0])) {
+        this.QUnit.config.modules[0].name === '') {
       globalSuite = this.convertModule(this.QUnit.config.modules[0])
       globalSuite.name = undefined
+
+      // The global tests suiteName must be undefined.
+      for (let test of globalSuite.tests) {
+        test.suiteName = undefined
+      }
+
+      modules = this.QUnit.config.modules.slice(1)
     } else {
       globalSuite = new Suite(undefined, [], [])
+      modules = this.QUnit.config.modules
     }
 
     // Build a list with all suites.
-    for (let module of this.QUnit.config.modules) {
-      if (!this.isGlobalModule(module)) {
-        suites.push(this.convertModule(module))
-      }
-    }
+    suites = modules.map(this.convertModule)
```

Comment:
Now that we don't need the initializer, let's remove the declaration above, and use `let suites =...` instead. Check if that can be applied for other vars as well.